src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/ISolutionExtensions.cs

```diff
@@ -79,4 +81,25 @@ public static TextDocument GetRequiredTextDocument(this Solution solution, Docum
 
     private static Exception CreateDocumentNotFoundException()
         => new InvalidOperationException(WorkspaceExtensionsResources.The_solution_does_not_contain_the_specified_document);
+
+#if !CODE_STYLE
+    public static Solution WithUpToDateSourceGeneratorDocuments(this Solution solution, ImmutableArray<ProjectId> projectIds)
```

Comment:
new helper for EnC to use.

Reply:
Any reason this can't be a proper method on Solution?

Reply:
Primarily because it didn't need any solution state.  and my (weak) preference is to try to keep those off instances to help make that clear.  i don't feel strongly.  this can move there if desired.